src/luarocks/build/rust-mlua.lua

```diff
@@ -35,7 +35,18 @@ function mlua.run(rockspec, no_install)
         return nil, "Lua version " .. lua_version .. " is not supported"
     end
 
-    local cmd = {"cargo build --release"}
+    local function rockspec_as_env_vars ()
+        local vars = {}
+        for _, key in ipairs({ "package", "version" }) do
+            table.insert(vars, ("LUA_ROCKSPEC_%s=%s"):format(key:upper(), fs.Q(rockspec[key])))
+        end
+        return vars
+    end
+
+    local cmd = {
+        "env " .. table.concat(rockspec_as_env_vars(), " "),
```

Comment:
I'm not 100% that this approach will work on windows.

There is a function [`execute_env`](https://github.com/luarocks/luarocks/blob/eae4e554d00000025c7d03e8a17bdb7245c7beff/src/luarocks/fs/lua.lua#L100) which we can use.
eg, replace:
```lua
if not fs.execute(table.concat(cmd, " ")) then
```
with 
```lua
if not fs.execute_env(envs, table.concat(cmd, " ")) then
```

and populate `envs` table

Reply:
Gotcha, coming right up...

Reply:
Nope that isn't right, stand by.

Reply:
There, that seems to work. I'm testing on Linux but this does what I expected again.